core/src/main/java/io/grpc/Metadata.java

```diff
@@ -430,7 +483,7 @@ public boolean equals(Object o) {
         return false;
       }
       Key<?> key = (Key<?>) o;
-      return name.equals(key.name);
+      return asciiName.equals(key.asciiName);
```

Comment:
Fixed.  This caused some other tests to show failures.  I have added another method called normalName(), and switched most of the code over.  This is a BEHAVIOR CHANGE, so I am not sure who should keep name().  Perhaps it should return the normalized name, and there should be a originalName() method?